### PR TITLE
chore: add sort option to more container list table columns

### LINKED ISSUE #7675: Container table sorting
### Is your enhancement related to a problem? Please describe

We provide sorting for most columns in all of our tables, but when I migrated the Containers page I only added sorting for the name column, but deferred it for others since there is a mix of object types.

### Describe the solution you'd like

Add sorting to all of the main columns, and confirm it works across different scenarios, e.g. standalone containers vs containers in a pod or compose group.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SoniaSandler**
### What does this PR do?

Adds more sort options to the container list table

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/7675

### How to test this PR?

- [] Tests are covering the bug fix or the new feature

Click on the table's column headers to check their sortings

### CODE REVIEW

File: packages/renderer/src/lib/container/ContainerList.svelte [hunk 1]
```diff
@@ -303,6 +304,11 @@ let statusColumn = new TableColumn<ContainerInfoUI | ContainerGroupInfoUI>('Stat
   align: 'center',
   width: '70px',
   renderer: ContainerColumnStatus,
+  comparator: (a, b) => {
+    const bStatus = ('status' in b ? b.status : 'state' in b ? b.state : '') ?? '';
```
Comment (jeffmaury) [inline packages/renderer/src/lib/container/ContainerList.svelte:308]: Can't it be rewrittten as bStatus = b.status ?? b.state ?? ''; ?

File: packages/renderer/src/lib/container/ContainerList.svelte [hunk 2]
```diff
@@ -313,11 +319,17 @@ let nameColumn = new TableColumn<ContainerInfoUI | ContainerGroupInfoUI>('Name',
...
 let envColumn = new TableColumn<ContainerInfoUI | ContainerGroupInfoUI>('Environment', {
   renderer: ContainerColumnEnvironment,
+  comparator: (a, b) => (a.engineType ?? '').localeCompare(b.engineType ?? ''),
 });
 
 let imageColumn = new TableColumn<ContainerInfoUI | ContainerGroupInfoUI>('Image', {
   width: '3fr',
   renderer: ContainerColumnImage,
+  comparator: (a, b) => {
+    const aImage = 'image' in a ? a.image : '';
```
Comment (jeffmaury) [inline packages/renderer/src/lib/container/ContainerList.svelte:329]: Same remark

**Review by jeffmaury — COMMENTED** [review]
Also would be good to have a unit test.

File: packages/renderer/src/lib/container/ContainerList.svelte (hunk 1 above)
Comment (SoniaSandler) [reply packages/renderer/src/lib/container/ContainerList.svelte:308]: @jeffmaury It gives me an error saying `Property 'status' does not exist on type 'ContainerGroupInfoUI | ContainerInfoUI'.
  Property 'status' does not exist on type 'ContainerInfoUI'` and `Property 'state' does not exist on type 'ContainerGroupInfoUI | ContainerInfoUI'.
  Property 'state' does not exist on type 'ContainerGroupInfoUI'`

File: packages/renderer/src/lib/container/ContainerList.svelte (hunk 2 above)
Comment (SoniaSandler) [reply packages/renderer/src/lib/container/ContainerList.svelte:329]: @jeffmaury same error as in the previous comment

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]
Tested and it works well and updates, thank you for this change!

Code and tests LGTM

**Pull request merged**